src/http/client.rs

```diff
@@ -20,14 +20,24 @@ impl Default for HttpClient {
 
 impl HttpClient {
   pub fn new(server: Server) -> Self {
-    let mut builder = Client::builder()
-      .pool_max_idle_per_host(200)
-      .tcp_keepalive(Some(Duration::from_secs(5)))
-      .timeout(Duration::from_secs(60))
-      .connect_timeout(Duration::from_secs(60))
-      .user_agent("Tailcall/1.0");
-
-    if let Some(ref proxy) = server.proxy {
+    let mut upstream_settings_serilaised: String = serde_json::to_string(&server.upstream).unwrap();
+    if upstream_settings_serilaised == "null" {
+      upstream_settings_serilaised = "{}".to_owned()
+    }
+    let upstream_settings: Upstream = serde_json::from_str(&upstream_settings_serilaised).unwrap();
```

Comment:
instead of doing this consider marking the `server.upstream` with `serde(default)` as well and implement Default trait for it

Reply:
@meskill done.

Reply:
you can fully get rid of conversions with serde_json and use just `server.upstream` - the defaults for it will be provided by serde on root parsing